(de)compiler fixes and improvements.

VladiStep opened the pull request:
## Description
1) Fixes #1612.
2) Added a "Check built-in functions arguments count" setting (similar to the "Feather" setting in GameMaker).
3) Updated some built-in function entries (argument types and count).
4) A lot of small optimizations for (de)compilation.
5) Removed the code that was overriding argument types for built-in functions (fixes #1610).

### Caveats
<!-- Any caveats, side effects or regressions of this PR -->

### Notes
<!-- Any notes or closing words -->

Siuhnexus commented:
Please add this to the tool, I had the annoying issue of wrong argument counts constantly when modding "Epic Auto Towers".

VladiStep commented:
It's not necessary anymore, because there is new "Underanalyzer" [de]compiler that addresses all of the issues - https://github.com/UnderminersTeam/UndertaleModTool/pull/2056.
